### DISCUSSION TITLE
Save manual mesh level to SD?

### DISCUSSION THREAD

**Discussion by Ali229** [Jyers/Marlin#583] [Ideas]
Is there a way to preserve the manual mesh level between firmware updates? I would like to:

1. Update firmware
2. Set factory settings (this will delete mesh)
3. Load mesh from SD card

**Comment by jcy12345** [comment]
This would be awesome. I'm updating to 1.3.1 right now and will lose my mesh. Not a HUGE deal as doing it manually takes a max of 10 minutes, but would be a nice to have.

**Comment by Jyers** [comment]
UBL offers this kind of functionality. Except instead of storing the mesh on the sd, it stores it in reserved portion of eeprom that doesn't get erased from flashings or resetting.

**Reply by Tapap84** [reply]
does manual mesh save there too? when i updated to 1.3.1 and i went to re input my mesh i had wrote down and i was noticing the mesh was already there

**Reply by Jyers** [reply]
It does not, but if you don't reset eeprom after a flash (which you should) there is a chance your mesh will still be there.

**Reply by Tapap84** [reply]
i have not done that, i didnt know it was a thing

**Reply by Jyers** [reply]
After a flash you generally want to go to the control menu and reset to defaults then store to clear eeprom

**Reply by Tapap84** [reply]
thanks. i was just googling because i didnt want to bother you with more questions.

**Reply by Jyers** [reply]
Its no problem at all, that's what I'm here for 😄
